include/bpf2c.h

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include <stdint.h>
+#include <stdbool.h>
+#include "ebpf_structs.h"
+#define UBPF_STACK_SIZE 512
+
+#define IMMEDIATE(X) (int32_t) X
+#define OFFSET(X) (int16_t) X
+#define POINTER(X) (uint64_t)(X)
+
+#define htobe16(X) swap16(X)
+#define htobe32(X) swap32(X)
+#define htobe64(X) swap64(X)
+
+#define htole16(X) (X)
+#define htole32(X) (X)
+#define htole64(X) (X)
+
+typedef struct _helper_function_entry
+{
+    uint64_t (*address)(uint64_t r1, uint64_t r2, uint64_t r3, uint64_t r4, uint64_t r5);
+    uint32_t helper_id;
+    const char* name;
+    bool tail_call;
+} helper_function_entry_t;
+
+typedef struct _map_entry
+{
+    void* address;
+    ebpf_map_definition_in_file_t definition;
+    const char* name;
+} map_entry_t;
+
```

Comment:
Please add a program_entry table as well which should store the addresses of the all the entry functions - and maybe the section name as well. This will be required for getting the program_info and attaching a program to a hook.

Reply:
> > I don't see any tests for this, can you add one? E.g., run droppacket.o through this and compare to known-good output? Or is there a way to take that output and somehow compile it back to ebpf byte code and verify it "matches" the original ebpf byte code (not sure if that's possible).
> 
> I was planning on using the test data from here: https://github.com/iovisor/ubpf/tree/master/tests
> 
> They include blocks of eBPF ASM and the expected return value.

Added tests for code gen based on the uBPF interpreter tests. About 90% of the tests pass, still need to investigate some of the failing tests.